oidc/oidc.go

```diff
@@ -378,6 +394,28 @@ type claimSource struct {
 	AccessToken string `json:"access_token"`
 }
 
+type stringAsBool bool
+
+func (sb *stringAsBool) UnmarshalJSON(b []byte) error {
+	var result bool
+	err := json.Unmarshal(b, &result)
+	if err == nil {
+		*sb = stringAsBool(result)
+		return nil
+	}
+	var s string
+	err = json.Unmarshal(b, &s)
+	if err != nil {
+		return err
+	}
+	result, err = strconv.ParseBool(s)
+	if err != nil {
+		return err
+	}
+	*sb = stringAsBool(result)
+	return nil
+}
```

Comment:
This might be simpler:

```go
func (sb *stringAsBool) UnmarshalJSON(b []byte) error {
	switch string(b) {
	case "true", `"true"`:
		*sb = stringAsBool(true)
	case "false", `"false"`:
		*sb = stringAsBool(false)
	default:
		return errors.New("wat")
	}
	return nil
}
```

Reply:
Good idea. Added as a separate commit to preserve commit history.